lib/screens/forgot_password/cubit/forgot_password_cubit.dart

```diff
@@ -12,37 +12,40 @@ class ForgotPasswordCubit extends Cubit<ForgotPasswordState> {
 
   ForgotPasswordCubit(this._forgotPasswordRepository) : super(ForgotEmailInitial());
 
-  void sendOTP(String email) async {
+  void sendOTP(String email, ForgotPasswordState state) async {
     try {
       emit(ForgotLoading());
       await _forgotPasswordRepository.sendOTP(email);
       emit(ForgotOTPInitial());
     } on NetworkException {
-      emit(ForgotNetworkError(S.networkException));
+      emit(ForgotNetworkError(S.networkException, state));
+    } on ResponseException catch (e) {
+      emit(ForgotOTPFailure(e.message, state));
     }
   }
 
-  Future<void> verifyOTP(String a) async {
+  Future<void> verifyOTP(String a, ForgotPasswordState state) async {
     emit(ForgotLoading());
     try {
-      bool b = await _forgotPasswordRepository.verifyOTP(a);
-      if (b) {
-        emit(ForgotResetInitial());
-      } else {
-        emit(ForgotOTPFailure());
-      }
+      await _forgotPasswordRepository.verifyOTP(a);
+      emit(ForgotResetInitial());
     } on NetworkException {
-      emit(ForgotNetworkError(S.networkException));
+      emit(ForgotNetworkError(S.networkException, state));
+    } on ResponseException catch (e) {
+      emit(ForgotOTPFailure(e.message, state));
     }
   }
 
-  Future<void> changePassword(String email, String otp, String password) async {
+  Future<void> changePassword(
+      String email, String otp, String password, ForgotPasswordState state) async {
```

Comment:
Since logic is supposed to be implemented in cubit, accept both passwords and check here if they are equal first before calling repository `changePassword` function.
If they are equal, pass the `newPassword` variable to the repository
If not, emit a `ForgotPasswordError` state and return.
```suggestion
  Future<void> changePassword(
      String email, 
      String otp, 
      String newPassword,
      String confirmPassword, 
      ForgotPasswordState state) async {
```

Reply:
This is remaining too.